### ISSUE TITLE
HTML indent calculation problem with comments

### ISSUE THREAD

**Issue by KristofMorva** [vuejs/eslint-plugin-vue#241]
If a HTML comment is the last entry in an indentation level, it expects the wrong level (always one indentation less).

For example the following example works as expected:
```
<div>
    <br>
    <!-- michalsnik rocks -->
    <br>
</div>
```
But if the comment is the last entry, we'll get a `Expected indentation of 0 spaces but found 4 spaces` error for the line of the comment:
```
<div>
    <br>
    <!-- michalsnik rocks -->
</div>
```
So it expects the comment to look like this:
```
<div>
    <br>
<!-- michalsnik rocks -->
</div>
```
I'd say, it's probably not what the author **usually** wants. I'd say it might be best to ignore indentation check for comments altogether, as we can't predict if the developer wants to comment out the last row, or write a comment for the closing tag.

**Comment by mysticatea** [comment]
Hmm, looks like a bug to me.

Indeed, core `indent` rule allows both indentations (same as the next token or the previous token) if the next token is a closing brace (`}`) for the following case:

```js
if (a) {
    foo()

// Description for the else
} else {
    foo()
}
```

If I clone the code to Vue.js template then:

```html
<div v-if="a">
    <some-item />

<!-- Description for the else. -->
</div><div v-else>
    <some-item />
</div>
```

I'm not sure whether this style is enough general or not.
If the style is not general, I think that we should fix this as a bug.

```html
<div>
  <some-item />
  <!-- this is correct -->
</div>
<div>
  <some-item />
<!-- this is incorrect -->
</div>
```

**Comment by mysticatea** [comment]
Maybe we should consider similar cases.

```html
<div>
    {{
        foo
        // This comment is adjusted to same as the next token }} currently.
        // I think this comment should be the same as the previous token.
    }}-{{
        bar
    }}
</div>
```

**Comment by michalsnik** [comment]
Alright, then I think we should just allow both prev and next token indentation for comments, and that's it.
